### DISCUSSION TITLE
Export to PDF

### DISCUSSION THREAD

**Discussion by IditYe** [react-page/react-page#920] [General] [answered]
Hi,
I've just found this project and it seems like a great one. Can someone please guide me how can I export a react page to a PDF file?
Thanks you.

**Comment by macrozone** [comment, marked as answer]
@IditYe there are two ways:

- process it through some html to pdf generator or run it through a headless browser and pupeteer. This way you can re-use any components and the result should look similar on pdf as on web. ReactPage works on server (at least for static output), so its possible to do that.
- You could use something like https://github.com/diegomura/react-pdf and  create a custom renderer for pdf, similar to the default static html/dom- output: https://github.com/react-page/react-page/blob/master/packages/editor/src/renderer/HTMLRenderer.tsx 
as you see it basically iterates over rows and cells and renders a Component for every cell that is defined by the plugin.id. you could replace these plugin definitions with one that has a Renderer that renders components from react-pdf. Thats probably easy for stuff like images, but a bit tricky for the rich-text-editor (which is using [slate](https://github.com/ianstormtaylor/slate/). Its probably best to understand the internal structure of slate and iterate over it and create text-blocks. 


The same applies to displaying content in react-native by the way. I have not done it yet, but it works in the same way. I am thinking about adding an example for react-native (and alongside develop a more general renderer), but currently i have no use case yet in a project .

**Reply by IditYe** [reply]
@macrozone Thank you very much.
